web-ui/src/main/resources/catalog/components/viewer/gfi/FeaturesLoader.js

```diff
@@ -91,18 +91,40 @@
       uuid = layer.get('metadataUuid');
     }
 
+    var infoFormat = layer.ncInfo ? 'text/xml' :
+                'application/vnd.ogc.gml';
+
+    //check if infoFormat is available in getCapabilities
+    if(layer.get('capRequest')) {
+      if($.inArray(infoFormat,
+          layer.get('capRequest').GetFeatureInfo.Format) == -1) {
```

Comment:
if `layer.get('capRequest').GetFeatureInfo` is `undefined` or `null` this can cause an error.

Reply:
in https://github.com/geonetwork/core-geonetwork/pull/2671/files#diff-f84568aec2e92783269ee51e149a71b4R928 `capRequest` can be set to `null`. That would cause an error.